type.go

```diff
@@ -0,0 +1,7 @@
+package dbin
+
+type Header struct {
+	Data        []byte
```

Comment:
The intent of this is not obvious due to its name. It is the raw bytes of the header ... (for me, Data and Header are opposite concepts!)

Maybe rename `Header.Data` to `Header.RawBytes` ?

Apart from this, LGTM!